integration/pvtdata/data_purge_test.go

```diff
@@ -0,0 +1,155 @@
+			SignaturePolicy:   `OR ('Org1MSP.member','Org2MSP.member', 'Org3MSP.member')`,
+			CollectionsConfig: CollectionConfig("collections_config1.json"),
+			Sequence:          "1",
+			InitRequired:      false,
+			Label:             "purgecc_label",
+		}
+
+		nwo.DeployChaincode(network, "testchannel", orderer, *chaincode)
+
+		org2Peer0 = network.Peer("Org2", "peer0")
+
+		_, cancel = context.WithTimeout(context.Background(), network.EventuallyTimeout)
+
+		marblechaincodeutil.AddMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-0", "color":"blue", "size":35, "owner":"tom", "price":99}`, org2Peer0)
+	})
+
+	AfterEach(func() {
+		cancel()
+
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+		if network != nil {
+			network.Cleanup()
+		}
+		os.RemoveAll(testDir)
+	})
+
+	// 6. The purge transaction takes effect only if the corresponding capability is set
+	//    - Add a few keys into a collection
+	//    - Issue a purge transaction without setting the new capability
+	//    - [Verify that the purge fails]
+	PIt("should fail with an error if the purge capability has not been enabled on the channel")
+
+	It("should prevent purged data being included in responses after the purge transaction has been committed", func() {
+		marblechaincodeutil.AssertPresentInCollectionM(network, channelID, chaincode.Name, `test-marble-0`, org2Peer0)
+		marblechaincodeutil.AssertPresentInCollectionMPD(network, channelID, chaincode.Name, "test-marble-0", org2Peer0)
+
+		marblechaincodeutil.PurgeMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-0"}`, org2Peer0)
+
+		marblechaincodeutil.AssertDoesNotExistInCollectionM(network, channelID, chaincode.Name, `test-marble-0`, org2Peer0)
+	})
+
+	// 1. User is able to submit a purge transaction that involves more than one keys
+	PIt("should accept multiple keys for purging in the same transaction")
+
+	// 2. The endorsement policy is evaluated correctly for a purge transaction under
+	//    different endorsement policy settings (e.g., collection level/ key-hash based)
+	//    Note: The endorsement policy level tests need not to be prioritized over other
+	//    behaviour, and they need not to be very exhaustive since they should be covered
+	//    by existing write/delete operations
+	PIt("should correctly enforce collection level endorsement policies")
+	PIt("should correctly enforce key-hash based endorsement policies")
+	PIt("should correctly enforce other endorsement policies (TBC)")
+
+	// 3. Data is purged on an eligible peer
+	//    - Add a few keys into a collection
+	//    - Issue a purge transaction for some of the keys
+	//    - Verify that all the versions of the intended keys are purged while the remaining keys still exist
+	//    - Repeat above to purge all keys to test the corner case
+	PIt("should remove all purged data from an eligible peer")
+
```

Comment:
This is a duplicate test of the test at line 106.

Reply:
Sure, we will have to figure out a way to do that in the integration test though! But, I am fine with merging having this for now.